### PR TITLE
GODRIVER-3061 Support Load Balancing in Docker Container

### PR DESCRIPTION
**Pull request by blink1073**
[GODRIVER-3061](https://jira.mongodb.org/browse/GODRIVER-3061)

## Summary
Support running the server and client separately in Docker containers, which allows us to start other services like the load balancer in parallel.

## Background & Motivation
We should support testing load balanced tests using docker containers.

### CODE REVIEW

File: mongo/client_test.go [hunk 1]
```diff
@@ -337,6 +337,9 @@ func TestClient(t *testing.T) {
 			if testing.Short() {
 				t.Skip("skipping integration test in short mode")
 			}
+			if len(os.Getenv("DOCKER_RUNNING")) > 0 {
```
Comment (qingyang-hu) [inline mongo/client_test.go:340]: Optional:
```suggestion
			if _, ok := os.LookupEnv("DOCKER_RUNNING"); ok {
```
or
```suggestion
			if os.Getenv("DOCKER_RUNNING") != "" {
```

File: mongo/with_transactions_test.go [hunk 2]
```diff
@@ -37,6 +38,9 @@ func TestConvenientTransactions(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping integration test in short mode")
 	}
+	if len(os.Getenv("DOCKER_RUNNING")) > 0 {
```
Comment (qingyang-hu) [inline mongo/with_transactions_test.go:41]: Optional:
```suggestion
	if _, ok := os.LookupEnv("DOCKER_RUNNING"); ok {
```
or
```suggestion
	if os.Getenv("DOCKER_RUNNING") != "" {
```

File: x/mongo/driver/operation/hello_test.go [hunk 3]
```diff
@@ -142,6 +143,10 @@ func TestAppendClientDriver(t *testing.T) {
 func TestAppendClientEnv(t *testing.T) {
 	clearTestEnv(t)
 
+	if len(os.Getenv("DOCKER_RUNNING")) > 0 {
```
Comment (qingyang-hu) [inline x/mongo/driver/operation/hello_test.go:146]: Optional:
```suggestion
	if _, ok := os.LookupEnv("DOCKER_RUNNING"); ok {
```
or
```suggestion
	if os.Getenv("DOCKER_RUNNING") != "" {
```

File: x/mongo/driver/operation/hello_test.go [hunk 4]
```diff
@@ -377,6 +382,10 @@ func TestAppendClientPlatform(t *testing.T) {
 func TestEncodeClientMetadata(t *testing.T) {
 	clearTestEnv(t)
 
+	if len(os.Getenv("DOCKER_RUNNING")) > 0 {
```
Comment (qingyang-hu) [inline x/mongo/driver/operation/hello_test.go:385]: Optional:
```suggestion
	if _, ok := os.LookupEnv("DOCKER_RUNNING"); ok {
```
or
```suggestion
	if os.Getenv("DOCKER_RUNNING") != "" {
```

File: x/mongo/driver/topology/server_test.go [hunk 5]
```diff
@@ -131,6 +131,9 @@ func TestServerHeartbeatTimeout(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping integration test in short mode")
 	}
+	if len(os.Getenv("DOCKER_RUNNING")) > 0 {
```
Comment (qingyang-hu) [inline x/mongo/driver/topology/server_test.go:134]: Optional:
```suggestion
	if _, ok := os.LookupEnv("DOCKER_RUNNING"); ok {
```
or
```suggestion
	if os.Getenv("DOCKER_RUNNING") != "" {
```

File: etc/docker_entry.sh [hunk 6]
```diff
@@ -1,17 +1,14 @@
 #!/usr/bin/env bash
 #
-# Entry point for Dockerfile for launching a server and running a go test.
+# Entry point for Dockerfile for launching running a go test.
```
Comment (qingyang-hu) [inline etc/docker_entry.sh:3]:
```suggestion
# Entry point for Dockerfile for running a go test.
```

**Review by qingyang-hu — COMMENTED** [review]
LGTM except a couple of trivial suggestions.

File: .evergreen/config.yml [hunk 7]
```diff
@@ -145,7 +145,7 @@ functions:
             # If this was a patch build, doing a fresh clone would not actually test the patch
             cp -R ${PROJECT_DIRECTORY}/ $DRIVERS_TOOLS
           else
-            git clone https://github.com/mongodb-labs/drivers-evergreen-tools.git $DRIVERS_TOOLS
+            git clone -b docker-update https://github.com/blink1073/drivers-evergreen-tools.git $DRIVERS_TOOLS
```
Comment (blink1073) [inline .evergreen/config.yml:148]: TODO: update before merging.

**Review by qingyang-hu — DISMISSED** [review]

**Comment by blink1073** [comment]
evergreen retry

**Pull request merged**